### ISSUE TITLE
Exception in one test crushes entire suit

### ISSUE THREAD

**Issue by kotslon** [theory/pgtap#68]
I'm trying to write my tests in xUnit style: i have many `test_*` functions and run them with `SELECT * FROM runtests();`.
But when DB structure changes or when i accidently introduce a syntax error in one of the tests, the entire suit crushes.

[Here](https://github.com/theory/pgtap/blob/master/sql/pgtap.sql.in#L6000) in the `_runner` function you just forward all exceptions occured in a test:

```sql
EXCEPTION WHEN raise_exception THEN
                IF SQLERRM <> '__TAP_ROLLBACK__' THEN
                    -- We didn't raise it, so propagate it.
                    RAISE EXCEPTION '%', SQLERRM;
                END IF;
            END;
```

Why? I have slightly modified that piece of `_runner`:

```sql
EXCEPTION
  WHEN raise_exception
    THEN
      IF SQLERRM <> '__TAP_ROLLBACK__'
        THEN
          -- We didn't raise it, so indicate, that something failed
          RETURN QUERY SELECT ok( FALSE, 'Unexpected exception' ) || E'\n' || diag(
            '        died: ' || SQLSTATE || ': ' || SQLERRM
          );
      END IF;
  WHEN OTHERS
    THEN
      -- We didn't raise it, so indicate, that something failed
        RETURN QUERY SELECT ok( FALSE, 'Unexpected exception' ) || E'\n' || diag(
          '        died: ' || SQLSTATE || ': ' || SQLERRM
      );
END;
```

And it works good for me now. But how was it intended to behave? I don't want to crash all my tests just because of one error. Maybe I'm doing something wrong?

**Comment by theory** [comment]
Hey @kotslon, I added a test to replicate this issue, then fixed it in 8fb3233 by making each test function a TAP-style subtest and capturing error messages before rolling back. No more error propagation and halting! Give it a shot.

**Comment by kotslon** [comment]
Glad to hear that! It's a great solution!

Sorry, i can't give it a shot now: i've finished that project half a year ago. But i'm still using postgres, so when i need test, i know where to go! Keep up the good work!
